src/query/functions/temporal/aggregation.go

```diff
@@ -0,0 +1,162 @@
+	CountTemporalType = "count_over_time"
+
+	// MinTemporalType calculates the minimum of all values in the specified interval
+	MinTemporalType = "min_over_time"
+
+	// MaxTemporalType calculates the maximum of all values in the specified interval
+	MaxTemporalType = "max_over_time"
+
+	// SumTemporalType calculates the sum of all values in the specified interval
+	SumTemporalType = "sum_over_time"
+
+	// StdDevTemporalType calculates the standard deviation of all values in the specified interval
+	StdDevTemporalType = "stddev_over_time"
+
+	// StdVarTemporalType calculates the standard variance of all values in the specified interval
+	StdVarTemporalType = "stdvar_over_time"
+)
+
+var (
+	aggFuncs = map[string]interface{}{
+		AvgTemporalType:    struct{}{},
+		CountTemporalType:  struct{}{},
+		MinTemporalType:    struct{}{},
+		MaxTemporalType:    struct{}{},
+		SumTemporalType:    struct{}{},
+		StdDevTemporalType: struct{}{},
+		StdVarTemporalType: struct{}{},
+	}
+)
+
+// NewAggOp creates a new base temporal transform with a specified node
+func NewAggOp(args []interface{}, optype string) (transform.Params, error) {
+	if _, ok := aggFuncs[optype]; !ok {
+		return emptyOp, fmt.Errorf("unknown aggregation type: %s", optype)
+	}
+
+	return newBaseOp(args, optype, newAggNode)
+}
+
+func newAggNode(op baseOp, controller *transform.Controller) Processor {
+	return &aggNode{
+		op:         op,
+		controller: controller,
+	}
+}
+
+type aggNode struct {
+	op         baseOp
+	controller *transform.Controller
+}
+
+func (a *aggNode) Process(values []float64) float64 {
+	switch a.op.operatorType {
+	case AvgTemporalType:
+		var mean, count float64
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				count++
+				mean += (v - mean) / count
+			}
+		}
+		return mean
+
+	case CountTemporalType:
+		var count float64
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				count++
+			}
+		}
+		return count
+
+	case MinTemporalType:
+		min := math.Inf(1)
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				min = math.Min(min, v)
+			}
+		}
+		return min
+
+	case MaxTemporalType:
+		max := math.Inf(-1)
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				max = math.Max(max, v)
+			}
+		}
+		return max
+
+	case SumTemporalType:
+		var sum float64
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				sum += v
+			}
+		}
+		return sum
+
+	case StdDevTemporalType:
+		var aux, count, mean float64
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				count++
+				delta := v - mean
+				mean += delta / count
+				aux += delta * (v - mean)
+			}
+		}
+		return math.Sqrt(aux / count)
+
+	case StdVarTemporalType:
+		var aux, count, mean float64
+		for _, v := range values {
+			if !math.IsNaN(v) {
+				count++
+				delta := v - mean
+				mean += delta / count
+				aux += delta * (v - mean)
+			}
+		}
+		return aux / count
+
+	default:
+		panic("unknown aggregation type")
```

Comment:
This doesn't seem right, but, on the other hand, if we get to this point, something is seriously wrong.

Reply:
Rather than doing this, you can have
`type aggFunc func(vals float64) float64` 
Then have these all be seperate functions, and have aggFuncs be a map of these, similar to how aggregation functions are done in aggregation/base.go

Reply:
Yeah, I thought it'd be a little messier, but on second thought, this is probably better. Will make the change.

Reply:
👍 
Another nice thing you get is that you don't need to check opType twice